File: 2021/11/20/main.go
package main

func singleNonDuplicate(nums []int) int {
	n := len(nums) - 1
	l, r := 0, n-1

	for l < r {
		m := l + (r-l)/2
		if m%2 == 1 {
			m--
		}

		if nums[m] != nums[m+1] {
			r = m
		} else {
			l = m + 2
		}
	}

	return nums[l]
}

File: 2021/11/20/main.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        n = len(nums) - 1
        l, r = 0, n-1

        while l < r:
            m = l + (r-l)//2
            if m%2 == 1:
                m -= 1

            if nums[m] != nums[m+1]:
                r = m
            else:
                l = m + 2

        return nums[l]